clippy_lints/src/uninit_vec.rs

```diff
@@ -211,9 +212,12 @@ fn extract_set_len_self<'tcx>(cx: &LateContext<'_>, expr: &'tcx Expr<'_>) -> Opt
         }
     });
     match expr.kind {
-        ExprKind::MethodCall(path, self_expr, [_], _) => {
+        ExprKind::MethodCall(path, self_expr, args, _) => {
```

Comment:
```suggestion
        ExprKind::MethodCall(path, self_expr, [arg], _) => {
```
You can match the single arg directly, because all `set_len` calls have exactly one argument. This will save you an `if let` in `is_literal_zero`.